tests/integration/security/file_mounted_certs/p2p_mtls_test.go

```diff
@@ -0,0 +1,440 @@
+			Ports:     []echo.Port{},
+			Subsets: []echo.SubsetConfig{{
+				Version: "v1",
+				// Set up custom annotations to mount the certs.
+				Annotations: echo.NewAnnotations().
+					Set(echo.SidecarVolume, `{"server-certs":{"secret":{"secretName":"test-cred"}},"client-certs":{"secret":{"secretName":"test-cred"}}}`).
+					Set(echo.SidecarVolumeMount, `{"server-certs":{"mountPath":"/server-certs"},"client-certs":{"mountPath":"/client-certs"}}`).
+					Set(echo.SidecarConfig, `{"controlPlaneAuthPolicy":"NONE","proxyMetadata":` + strings.Replace(ProxyMetadataJson, "\n", "", -1) + `}`),
+			}},
+		}).
+		With(&internalServer, echo.Config{
+			Service:   "server",
+			Namespace: appsNamespace,
+			Ports: []echo.Port{
+				{
+					Name:         "http",
+					Protocol:     protocol.HTTP,
+					ServicePort:  8443,
+					InstancePort: 8443,
+					TLS:          false,
+				},
+			},
+			Subsets: []echo.SubsetConfig{{
+				Version:     "v1",
+				// Set up custom annotations to mount the certs.
+				Annotations: echo.NewAnnotations().
+					Set(echo.SidecarVolume, `{"server-certs":{"secret":{"secretName":"test-cred"}},"client-certs":{"secret":{"secretName":"test-cred"}}}`).
+					Set(echo.SidecarVolumeMount, `{"server-certs":{"mountPath":"/server-certs"},"client-certs":{"mountPath":"/client-certs"}}`).
+					Set(echo.SidecarConfig, `{"controlPlaneAuthPolicy":"NONE","proxyMetadata":` + strings.Replace(ProxyMetadataJson, "\n", "", -1) + `}`),
+			}},
+		}).
+		BuildOrFail(t)
+
+	return internalClient, internalServer, appsNamespace
+}
+
+
+func clusterName(target echo.Instance, port echo.Port) string {
+	cfg := target.Config()
+	return fmt.Sprintf("outbound|%d||%s.%s.svc.%s", port.ServicePort, cfg.Service, cfg.Namespace.Name(), cfg.Domain)
+}
+
+
+func getMetricForCluster(metrics []*dto.Metric, clusterName string) *dto.Metric {
+	for _, metric := range metrics {
+		for _, label := range metric.Label {
+			if (*label.Name == "cluster_name" && *label.Value == clusterName) {
+				return metric
+			}
+		}
+	}
+	return nil
+}
+
+func getMetricForListener(metrics []*dto.Metric, listenerName string) *dto.Metric {
+	for _, metric := range metrics {
+		for _, label := range metric.Label {
+			if (*label.Name == "listener_address" && *label.Value == listenerName) {
+				return metric
+			}
+		}
+	}
+	return nil
+}
+
+
+func verifyUsingFileBasedCerts(t *testing.T, client echo.Instance, server echo.Instance) {
+	// Verify client side configs
+	clientWorkloads, _ := client.Workloads()
+	if clientWorkloads[0].Sidecar() == nil {
+		t.Fatalf("Sidcecar is expected, but was not found")
+	}
+
+	configDump, err := clientWorkloads[0].Sidecar().Config()
+	if err != nil {
+		t.Fatalf("Unable to retrieve config_dump from the client. %v", err)
+	}
+
+	clusterName := clusterName(server, echo.Port{ServicePort: 8443})
+
+	validator := structpath.ForProto(configDump)
+	destCluster := validator.Select("{.configs[*].dynamicActiveClusters[?(@.cluster.name == '%s')]}", clusterName)
```

Comment:
This validation of config is not needed. We do have unit tests that validate this behaviour. We just want e->e verification here.

Reply:
Wanted to have this just to be on the safe side. ie - to know for sure that filebased certs were configured/used